packages/core/src/StateMachine.ts

```diff
@@ -223,27 +229,18 @@ export class StateMachine<
   public resolveState(
     state: State<TContext, TEvent, TResolvedTypesMeta>
   ): typeof state {
-    const configuration = Array.from(
-      getConfiguration(getStateNodes(this.root, state.value))
+    const configurationSet = getConfiguration(
+      getStateNodes(this.root, state.value)
     );
+    configurationSet.add(this.root); // TODO: fix in getStateNodes or getConfiguration
```

Comment:
This was fixed in #3551

Reply:
Where? How? The stacked PR split here doesn't make it easier to review stuff like this.

Reply:
I unstacked it and merged that into here